test_validator.py

```diff
@@ -26,6 +26,54 @@ def test_validate(self):
         self.assertEqual(True, self.validator.validate(book1))
         self.assertAlmostEqual(False, self.validator.validate(book2))
 
+    def test_validate_email_success(self):
+        validator = Validator()
+        validator.add({'email': [Required(), Email()]})
+        data = {
+            "email": "test@test.com"
+        }
+        result = validator.validate(data)
+        self.assertEqual(True, result)
+        self.assertEqual(0, len(validator.errors))
+
+        data = {
+            "email": "test@triangle.software"
+        }
+        result = validator.validate(data)
+        self.assertEqual(True, result)
+
+    def test_validate_complex_data(self):
+        validator = Validator()
+        validator.add({'name': [Required(), Length(min=3, max=50)]})
+        validator.add({'email': [Required(), Email()]})
+        validator.add({'admin_email': [Email()]})
+        data = {
+            "name": "Test",
+            "email": "test@triangle.software",
+            "admin_email": "admin@triangle.software"
+        }
+        result = validator.validate(data)
+        self.assertEqual(True, result)
+        self.assertEqual(0, len(validator.errors))
+
+    def test_validate_data_empty(self):
+        validator = Validator()
+        validator.add({'email': [Required(), Email()]})
+        data = {}

```

Comment:
Why validator.errors for empty data must be 2?

Reply:
2 validation rules will be failed on empty data: Required, Email